raiden/tests/smart_contracts/test_channel_manager.py

```diff
@@ -134,14 +140,72 @@ def test_channelmanager(tester_state, tester_token, tester_events,
         'settleTimeout': settle_timeout,
     }
 
-    # uncomment private in function to run test
-    # assert channel_manager.numberOfItems(netting_channel_creator1) == 2
-    # assert channel_manager.numberOfItems(sha3('address1')[:20]) == 1
-    # assert channel_manager.numberOfItems(sha3('iDontExist')[:20]) == 0
-    # vs = sorted((sha3('address1')[:20], sha3('address2')[:20]))
-    # k0 = channel_manager.key(sha3('address1')[:20], sha3('address2')[:20])
-    # assert k0 == sha3(vs[0] + vs[1])
-    # k1 = channel_manager.key(sha3('address2')[:20], sha3('address1')[:20])
-    # assert k1 == sha3(vs[0] + vs[1])
-    # with pytest.raises(TransactionFailed):
-    #    channel_manager.key(sha3('address1')[:20], sha3('address1')[:20])
+
+def test_deleteChannel(tester_state, tester_channelmanager, tester_channels, settle_timeout,
+                       netting_channel_abi):
+    privatekey0_raw, privatekey1_raw, nettingchannel, channel0, _ = tester_channels[0]
+
+    privatekey0 = PrivateKey(privatekey0_raw, ctx=GLOBAL_CTX, raw=True)
+    address0 = privatekey_to_address(privatekey0_raw)
+    address1 = privatekey_to_address(privatekey1_raw)
+
+    # We need to close the channel before it can be deleted, to do so we need
+    # one transfer to call closeSingleTransfer(0
+    transfer_amount = 10
+    identifier = 1
+    direct_transfer = channel0.create_directtransfer(
+        transfer_amount,
+        identifier,
+    )
+    direct_transfer.sign(privatekey0, address0)
+    direct_transfer_data = str(direct_transfer.packed().data)
+
+    should_be_nonce = nettingchannel.opened(sender=privatekey0_raw) * (2**32)
+    should_be_nonce_plus_one = (nettingchannel.opened(sender=privatekey0_raw) + 1) * (2**32)
+    assert should_be_nonce <= direct_transfer.nonce
```

Comment:
assert should_be_nonce <= direct_transfer.nonce < should_be_nonce_plus_one is much better